genic/main.c

```diff
@@ -29,13 +30,22 @@ int main(int argc, char **argv)
 
   walltime_init(&CT);
   mymalloc_init(MaxMemSizePerNode);
-  const double meanspacing = Box / Ngrid;
-  const double shift_gas = -0.5 * (CP.Omega0 - CP.OmegaBaryon) / CP.Omega0 * meanspacing;
-  const double shift_dm = +0.5 * CP.OmegaBaryon / CP.Omega0 * meanspacing;
   int64_t TotNumPart = (int64_t) Ngrid*Ngrid*Ngrid;
 
+  init_cosmology(&CP, InitTime);
+
   initialize_powerspectrum(ThisTask, InitTime, UnitLength_in_cm, &CP, &PowerP);
   petapm_init(Box, Nmesh, 1);
+  /*Initialise particle spacings*/
+  const double meanspacing = Box / Ngrid;
+  const double shift_gas = -ProduceGas * 0.5 * (CP.Omega0 - CP.OmegaBaryon) / CP.Omega0 * meanspacing;
+  double shift_dm = ProduceGas * 0.5 * CP.OmegaBaryon / CP.Omega0 * meanspacing;
+  double shift_nu = 0;
+  if(!ProduceGas && NGridNu > 0) {
```

Comment:
Does this suggest ProduceGas and NGridNu cannot be used the same time?

Reply:
No, that's fine. In that gas and DM are shifted, and NU sits in the middle on the grid points. As long as they do not initially have the same positions it is ok.